example/flutter/objectbox_demo/lib/main.dart

```diff
@@ -43,143 +49,169 @@ class MyHomePage extends StatefulWidget {
   _MyHomePageState createState() => _MyHomePageState();
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  final _noteInputController = TextEditingController();
+class ViewModel {
   Store _store;
   Box<Note> _box;
-  List<Note> _notes = [];
+  Query<Note> _query;
+
+  ViewModel(Directory dir) {
+    _store = Store(getObjectBoxModel(), directory: dir.path + '/objectbox');
+    _box = Box<Note>(_store);
+
+    final dateProp = Note_.date;
+    final dummyCondition = dateProp.greaterThan(0);
+
+    _query = _box
+        .query(dummyCondition)
+        .order(dateProp, flags: Order.descending)
+        .build();
+  }
+
+  void addNote(Note note) => _box.put(note);
+
+  void removeNote(Note note) => _box.remove(note.id);
+
+  get queryStream => _query.findStream();
+
+  get allNotes => _query.find();
+
+  void dispose() {
+    _query.close();
+    _store.unsubscribe();
```

Comment:
I wonder if this should be part of `close()`. This feels like something very easy to forget.

Reply:
It's called by `close` iirc. I didn't want to expose the BLoC/VM's internals to the Widgets.

Reply:
Oh, sorry, should have been more clear: was talking about not even exposing `unsubscribe()` to the user and doing it as part of `close()`.

Reply:
Good point.